storage/database.go

```diff
@@ -356,17 +551,61 @@ func (d *db) namespaceFor(namespace ts.ID) (databaseNamespace, error) {
 	return n, nil
 }
 
-func (d *db) getOwnedNamespaces() []databaseNamespace {
-	d.RLock()
+func (d *db) ownedNamespacesWithLock() []databaseNamespace {
 	namespaces := make([]databaseNamespace, 0, len(d.namespaces))
 	for _, n := range d.namespaces {
 		namespaces = append(namespaces, n)
 	}
-	d.RUnlock()
 	return namespaces
 }
 
+func (d *db) GetOwnedNamespaces() []databaseNamespace {
+	d.RLock()
+	defer d.RUnlock()
+	return d.ownedNamespacesWithLock()
+}
+
 func (d *db) nextIndex() uint64 {
 	created := atomic.AddUint64(&d.created, 1)
 	return created - 1
 }
+
+type tsIDs []ts.ID
+
+func (t tsIDs) String() (string, error) {
+	if len(t) == 0 {
+		return "[]", nil
+	}
+	var buf bytes.Buffer
```

Comment:
nit: Considering you return `[]` if `len(t) == 0` perhaps would be best to prefix and suffix if you have real values with `[` and `]`?  i.e. `buf.WriteRune('[')` at beginning and `buf.WriteRune(']')` at end?

Reply:
sure thing

Reply:
done